.travis.yml

```diff
@@ -20,11 +23,14 @@ matrix:
     - env: GCC=gcc-6     CXX=g++-6       LIBTYPE=system  PACKAGES=g++-6     PPA=ubuntu-toolchain-r/test
     - env: GCC=clang-4.0 CXX=clang++-4.0 LIBTYPE=system  PACKAGES="clang-4.0 llvm-4.0-dev" PPA=ubuntu-toolchain-r/test LLVM=llvm-toolchain-trusty-4.0
     - env: GCC=clang-5.0 CXX=clang++-5.0 LIBTYPE=bundled CMAKE_OPT="-DWITH_PAM=1" PACKAGES="clang-5.0 llvm-5.0-dev" LLVM=llvm-toolchain-trusty-5.0
+    - os: osx
+      osx_image: xcode9.3
+      env: LIBTYPE=bundled GCC=clang CXX=clang++ MAINTAINER_MODE=OFF
```

Comment:
Possible to do brew install openssl@1.0 and then LIBTYPE=system for faster builds. CMake needs -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl/ then

Reply:
The question is, which would be more realistic/better? For the other builds, we have builds wiith bundled and system libraries, but with osx, I wanted to avoid that, as the 5.7 build is slow anyway.

I went with bundled, and didn't even try it with the system version, because I though that uses a larger percentage of the codebase - but I'm not using osx for development, so you probably have a better idea about this.

Reply:
There is no point in covering larger percentage of codebase since no official support: the last thing I want is to fix platform bugs in shipped 3rd party libraries, which we don't do much on Linux neither. Hence, build speed and not build breaking due to 3rd party libs is more important here, IMHO

Reply:
This seems to be `clang-4.0` issue, not macOS issue.
I have the same error as macOS on Ubuntu with `clang-4.0` at https://travis-ci.org/inikep/percona-server/jobs/337929558:
```error: invalid value 'gnu++03' in '-std=gnu++03'
make[2]: *** [sql/CMakeFiles/gen_lex_hash.dir/genThis seems to be `clang-4.0` issue, not macOS issue.
_lex_hash.cc.o] Error 1
make[1]: *** [sql/CMakeFiles/gen_lex_hash.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....```
```

Reply:
@inikep - right, with the changes laurynas suggested in the 5.7 review, this is also fixed

@laurynas-biveinis - I tested this, and looks like using brew / the system libraries doesn't result in a speed improvement - maybe because the brew install task itself takes time. On the other hand, it complicates the config because of the required additional cmake option.